### PR TITLE
Move snapshot doc to asciidoc and add plugin installation section

### PR DESCRIPTION
**Pull request by pebrc**
* Convert the existing snapshot documentation to asciidoc 
* Add a section about plugin installation now that #35 is done

Relevant for #1051

### CODE REVIEW

File: docs/snapshots.asciidoc [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+To use your own custom image with all necessary plugins pre-installed use an Elasticsearch resource like this one:
+
+[source,yaml]
+----
+apiVersion: elasticsearch.k8s.elastic.co/v1alpha1
+kind: Elasticsearch
+metadata:
+  name: elasticsearch-sample
+spec:
+  version: "7.1.0"
```
Comment (thbkrkr) [inline docs/snapshots.asciidoc]: Nit: We could save 2 chars? We don't use double quotes in the quickstart :)

File: docs/snapshots.asciidoc [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+. Make an HTTP request against the appropriate endpoint, using a daily snapshot naming format. Elasticsearch credentials are mounted as a volume into the job's pod:
++
+[source,yaml]
+----
+# snapshotter.yml
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: my-cluster-snapshotter
```
Comment (thbkrkr) [inline docs/snapshots.asciidoc]: The Elasticsearch is named `elasticsearch-sample` but this job `my-cluster`. I think I'd like to reuse the same name.

File: docs/snapshots.asciidoc [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+The examples below use the https://www.elastic.co/guide/en/elasticsearch/plugins/master/repository-gcs.html[Google Cloud Storage Repository Plugin].
+
+For more information on Elasticsearch snapshots, see https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-snapshots.html[Snapshot and Restore].
+
+[id="{p}-install-plugin"]
+== Install the Storage Repository Plugin
+
+You can install the storage repository plugin either by using a custom image or by adding your own init container which
+installs the plugin at pod creation time.
+
```
Comment (alaudazzi) [inline docs/snapshots.asciidoc]: => suggested rewording:

`To install the storage repository plugin, you can either use a custom image or add your own init container which installs the plugin at pod creation time.`

File: docs/snapshots.asciidoc [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+To use your own custom image with all necessary plugins pre-installed use an Elasticsearch resource like this one:
```
Comment (alaudazzi) [inline docs/snapshots.asciidoc]: => suggested rewording:

To use your own custom image with all necessary plugins pre-installed, use an Elasticsearch resource like the following one:

File: docs/snapshots.asciidoc [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+metadata:
+  name: elasticsearch-sample
+spec:
+  version: 7.1.0
+  image: your/custom/image:7.1.0
+  nodes:
+  - nodeCount: 1
+----
+
+Or alternatively install the plugin at pod creation time by using an init container:
```
Comment (alaudazzi) [inline docs/snapshots.asciidoc]: Remove `or`

Comment (alaudazzi) [inline docs/snapshots.asciidoc]: Remove `or`.

**Pull request merged**